### PR TITLE
Home page ViewModel

### PR DESCRIPTION
**Pull request by TaiHaDev**
## Describe your changes

- Added logic for shifts fetching and dummy data for testing and visualisation
- Merged main branch to get the view and made it compatible
image

### CODE REVIEW

File: fireapp/lib/presentation/volunteer_home/volunteer_home_view_model.dart [hunk 1]
```diff
@@ -25,17 +27,28 @@ class VolunteerHomeViewModel extends FireAppViewModel {
 
   VolunteerHomeViewModel(
       this._authenticationRepository, this._shiftsRepository);
-
-  // write your functions and logic etc
-
-  //TODO retrieve shifts
   Future<void> fetchShifts() async {
+    _loadingState.add(RequestState.loading());
```
Comment (emilymclean) [inline fireapp/lib/presentation/volunteer_home/volunteer_home_view_model.dart]: You should be setting `_shifts` to `RequestState.loading()`, not this. Each RequestState stream is supposed to represent the lifecycle of a *single* operation (fetching some kind of data, posting a form, etc), so splitting it across multiple states isn't good practice. `_loadingState` isn't actually used here besides this, so I'd appreciate it if you just deleted it completely.

File: fireapp/lib/presentation/volunteer_home/volunteer_home_view_model.dart [hunk 2]
```diff
@@ -25,17 +27,28 @@ class VolunteerHomeViewModel extends FireAppViewModel {
...
+    try {
+      var userID =
+          (await _authenticationRepository.getCurrentSession())?.userId;
+      // Check if userId is null and throw an exception if it is
+      if (userID == null) {
+        throw Exception(
+            'User ID is null. Cannot fetch shift without a valid user ID.');
+      }
+      var shifts =
+      await _shiftsRepository.getVolunteerShifts(userID);
```
Comment (emilymclean) [inline fireapp/lib/presentation/volunteer_home/volunteer_home_view_model.dart]: Please indent this

**Review by emilymclean — CHANGES_REQUESTED** [review]
Looks good, just a couple of quick changes before I approve.

**Comment by TaiHaDev** [comment]
Hi @BenMMcLean, I have committed your suggested changes. Can you take a look?👀

**Review by emilymclean — DISMISSED** [review]

**Review by emilymclean — APPROVED** [review]

**Pull request merged**